pkg/autoupdate/tuf/ci/tuf_server.go

```diff
@@ -17,6 +15,34 @@ import (
 	"github.com/theupdateframework/go-tuf"
 )
 
+var (
+	//go:embed testdata/darwin_launcher.tar.gz
+	darwinLauncher []byte
+	//go:embed testdata/darwin_osqueryd.tar.gz
+	darwinOsqueryd []byte
+	//go:embed testdata/linux_launcher.tar.gz
+	linuxLauncher []byte
+	//go:embed testdata/linux_osqueryd.tar.gz
+	linuxOsqueryd []byte
```

Comment:
These were compiled from the same source? Does it make sense to embed the same thing twice, and slightly reduce the git checkout size?

Reply:
Yes, compiled from the same source, but I renamed the executable for each (launcher/launcher.exe vs osqueryd/osqueryd.exe) -- the executable's name has to match the given TUF target. Not sure there's a simple way around that in the tests, but maybe there's something I'm not thinking of?

Reply:
I'm not sure it's worth it, but embed is just about stashing the binary bits in a go structure. How you extract them is up to you. You have:
https://github.com/kolide/launcher/pull/1401/files#diff-114d276c4d96fce71399f71f2f007b77eaddd1a3b4109e1b8430cc9d7f4c8ec0R21-R28
```go
func getTarballContents(t *testing.T, binary string) []byte {
	tarballName := fmt.Sprintf("testdata/%s_%s.tar.gz", runtime.GOOS, binary)

	contents, err := testTarballs.ReadFile(tarballName)
	require.NoError(t, err)

	return contents
}
```

But if you wanted to merge binaries, I think you'd do something like:
```
func getTarballContents(t *testing.T) []byte {
	tarballName := fmt.Sprintf("testdata/%s.tar.gz", runtime.GOOS)

	contents, err := testTarballs.ReadFile(tarballName)
	require.NoError(t, err)

	return contents
}
```

All the naming is in `f.Write(getTarballContents(t, b))`

Reply:
It's not just the tarball that has to be renamed, though -- the binary inside the tarball that needs to be renamed also?

Reply:
_oh_. That makes sense. And the whole point of this is to pre-generate the tarballs. :hides: